generators/chipyard/src/main/scala/IOBinders.scala

```diff
@@ -65,74 +77,146 @@ class ComposeIOBinder[T](fn: => (Clock, Bool, Bool, T) => Seq[Any])(implicit tag
+    Seq((ports2d.flatten, ioCells2d.flatten, Some(harnessFn)))
+  }
+})
+// DOC include end: WithGPIOTiedOff
+
+class WithUARTAdapter extends OverrideIOBinder({
+  (system: HasPeripheryUARTModuleImp) => {
+    val (ports, ioCells2d) = AddIOCells.uart(system.uart)
+    val harnessFn = (th: chipyard.TestHarness) => { UARTAdapter.connect(ports)(system.p); Nil }
+    Seq((ports, ioCells2d.flatten, Some(harnessFn)))
+  }
 })
 
 class WithSimBlockDevice extends OverrideIOBinder({
-  (c, r, s, top: CanHavePeripheryBlockDeviceModuleImp) => top.connectSimBlockDevice(c, r); Nil
+  (system: CanHavePeripheryBlockDeviceModuleImp) => system.connectSimBlockDevice(system.clock, system.reset.asBool); Nil
 })
 
 class WithBlockDeviceModel extends OverrideIOBinder({
-  (c, r, s, top: CanHavePeripheryBlockDeviceModuleImp) => top.connectBlockDeviceModel(); Nil
+  (system: CanHavePeripheryBlockDeviceModuleImp) => system.connectBlockDeviceModel(); Nil
 })
 
 class WithLoopbackNIC extends OverrideIOBinder({
-  (c, r, s, top: CanHavePeripheryIceNICModuleImp) => top.connectNicLoopback(); Nil
+  (system: CanHavePeripheryIceNICModuleImp) => system.connectNicLoopback(); Nil
 })
 
 class WithSimNIC extends OverrideIOBinder({
-  (c, r, s, top: CanHavePeripheryIceNICModuleImp) => top.connectSimNetwork(c, r); Nil
-})
-
-class WithUARTAdapter extends OverrideIOBinder({
-  (c, r, s, top: HasPeripheryUARTModuleImp) => {
-    val defaultBaudRate = 115200 // matches sifive-blocks uart baudrate
-    top.uart.zipWithIndex.foreach{ case (dut_io, i) =>
-      val uart_sim = Module(new UARTAdapter(i, defaultBaudRate)(top.p))
-      uart_sim.io.uart.txd := dut_io.txd
-      dut_io.rxd := uart_sim.io.uart.rxd
-    }
-    Nil
-  }
+  (system: CanHavePeripheryIceNICModuleImp) => system.connectSimNetwork(system.clock, system.reset.asBool); Nil
 })
 
 // DOC include start: WithSimAXIMem
 class WithSimAXIMem extends OverrideIOBinder({
-  (c, r, s, top: CanHaveMasterAXI4MemPortModuleImp) => top.connectSimAXIMem(); Nil
+  (system: CanHaveMasterAXI4MemPortModuleImp) => system.connectSimAXIMem(); Nil
 })
 // DOC include end: WithSimAXIMem
 
 class WithBlackBoxSimMem extends OverrideIOBinder({
-  (clock, reset, _, top: CanHaveMasterAXI4MemPortModuleImp) => {
-    (top.mem_axi4 zip top.outer.memAXI4Node).foreach { case (io, node) =>
-      val memSize = top.p(ExtMem).get.master.size
-      val lineSize = top.p(CacheBlockBytes)
+  (system: CanHaveMasterAXI4MemPortModuleImp) => {
+    (system.mem_axi4 zip system.outer.memAXI4Node).foreach { case (io, node) =>
+      val memSize = system.p(ExtMem).get.master.size
+      val lineSize = system.p(CacheBlockBytes)
       (io zip node.in).foreach { case (axi4, (_, edge)) =>
         val mem = Module(new SimDRAM(memSize, lineSize, edge.bundle))
         mem.io.axi <> axi4
-        mem.io.clock := clock
-        mem.io.reset := reset
+        mem.io.clock := system.clock
+        mem.io.reset := system.reset
       }
     }; Nil
```

Comment:
Nit: Drop the semicolon and put the Nil on another line, unless you're doing a one-liner.

Reply:
Yeah, I think it was just a copy/paste artifact. I will fix it if I end up making another commit, otherwise if you want to fix it in your PR that would help.